chips/cc26x0/src/udma.rs

```diff
@@ -2,42 +2,72 @@
 /// # Micro Direct Memory Access for the TI CC26x0 Microcontroller
 ///
 
-use kernel;
 use kernel::common::regs::{ReadOnly, ReadWrite, WriteOnly};
+use kernel::ReturnCode;
+use prcm;
 
 pub const UDMA_BASE: usize = 0x4002_0000;
-pub const UDMA_CONFIG_BASE: usize = 0x2000_0400;
 
 #[repr(align(1024))]
 pub struct ControlTable {
-    config_array: [u8; 1024],
+    config_array: [DMAConfig; 32]
 }
 
-/*
+#[derive(Copy, Clone)]
 struct DMAConfig {
     source_ptr: usize,
     dest_ptr: usize,
-    control: usize,
+    control: ReadWrite<u32,DMATableControl::Register>,
```

Comment:
Having trouble here- I'd like to implement the control pointer as a register, since it has a register-like structure. Unfortunately, I need to implement `copy` and `clone` in order to get this to work in its current form. That means I need to modify the `mod.rs` file in the kernel (which I don't want to do just for this...)

Reply:
Hrm... that's a bummer. I don't see any harm in adding `Copy` and `Clone` to `ReadWrite` (since it's really just a wrapper around an `IntLike` (which already needs to implement `Copy` and `Clone`)